src/utils/editor.ts

```diff
@@ -251,6 +253,18 @@ export const extractMetadata = (body: string, thumbUrl?: string) => {
     out.users = matchedUsers.slice(0, 10); // return only first 10 users
   }
 
+  if (out.image && fetchRatios) {
+    out.image_ratios = await Promise.all(out.image.slice(0, 5).map((url) => {
+      return new Promise((resolve) => {
+        Image.getSize(url, (width, height) => {
+          resolve(width / height)
+        }, () => resolve(0))
```

Comment:
@noumantahir If fail, `resolve(1)` perhaps instead of `0`? Also 2nd slice not necessary?

Reply:
@feruzm it's intentional, defaulting to 1 would imply a square image while 0 would give a meaning it's ratio is undefined after front-end will fallback to old way to setting ratio.... but if we set it to undefined then array itself would be messed with and might interfere with cross matching `image` and `image_ratios` arrays.

Suggest otherwise if you find this approach troublesome.